### PR TITLE
Pull in new "Mandated Category" field and align with skeleton-python-library

### PR DESCRIPTION
**Pull request by dav3r**
## 🗣 Description

The "Mandated Category" field was recently added to assessment data.  This PR adds the new field to the list of fields imported to our mongoDB from the assessment JSON file.  Note that since older assessments do not have the "Mandatory Category" field, we treat it as optional during the import (i.e. it's possible for this field to have a value of `null`). 

I also decided to make [`skeleton-python-library`](https://github.com/cisagov/skeleton-python-library) the "base" skeleton for this project, so that future upstream improvements can easily be pulled in here.

The "Mandated Category" code change can be seen in https://github.com/cisagov/assessment-data-import-lambda/commit/b1a15d21475534730ffe274fdddac3b707bd4914, while all of the other commits are related to pulling in [`skeleton-python-library`](https://github.com/cisagov/skeleton-python-library) code and appropriately modifying it for this project.

## 💭 Motivation and Context

This change was needed after [JRD-663](https://jira.ncats.cyber.dhs.gov/browse/JRD-663) added the "Mandated Category" field to the assessment workflow.

## 🧪 Testing

Testing of the "Mandated Category" code change was done via ocular review, due to its simplicity.  The remaining changes were tested by running `pytest`, `pre-commit`, and `docker-compose`.

Once this change is deployed, the results will be verified in Production to ensure correctness.

## 🚥 Types of Changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (causes existing functionality to change)

## ✅ Checklist

- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation (Just not this documentation; stay tuned for an upcoming PR in https://github.com/cisagov/ncats-data-dictionary).
- [ ] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -1,5 +1,7 @@
-.python-version
-.DS_Store
 *.egg-info
 __pycache__
+.python-version
+.coverage
+.pytest_cache
+.DS_Store
```
Comment (jsf9k) [inline .gitignore:6]: Apple!

**Review by jsf9k — APPROVED** [review]
Nice work @dav3r, but delete the `.travis.yml` file!

**Review by mcdonnnj — APPROVED** [review]
Looking good and I appreciate basing on `skeleton-python-library` to streamline future upkeep. I'm assuming we should remove `.travis.yml` as part of transitioning to GitHub Actions. Strong work 💪

**Comment by dav3r** [comment]
Thanks @jsf9k and @mcdonnnj for catching the `.travis.yml` file - I removed it in https://github.com/cisagov/assessment-data-import-lambda/pull/3/commits/f480bcd998d3e997556083bafc2c341b3db8caaf

**Review by felddy — APPROVED** [review]
Ocular review... 🙄 

Looks good.  Thanks for syncing with the py skeleton.  💪

**Comment by felddy** [comment]
@dav3r,  I know you are keen on the QA and wanted to make sure you saw that the `build` action has a spiffy feature:  You can [download the artifact from the build process](https://github.com/cisagov/assessment-data-import-lambda/pull/3/checks?check_run_id=335010687) to check locally.  In case you wanted to do "enhanced ocular reviews" 😝 .

image

**Comment by felddy** [comment]
Still looks good to me!
🍺 🍻🍺 🍻🍺 🍻🍺 🍻🍺 🍻🍺 🍻 🍺

**Pull request merged**